src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterSpec.java

```diff
@@ -0,0 +1,55 @@
+package io.stargate.sgv2.jsonapi.api.model.command.clause.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.deserializers.FilterClauseDeserializer;
+import java.util.Objects;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/**
+ * Intermediate lightly-processed container for JSON that specifies a {@link FilterClause}, and
+ * allows for lazy deserialization into a {@link FilterClause}.
```

Comment:
From discussions, is there an abstraction here we can use so this class is re-usable for different Objects ?

Reply:
It is definitely doable. The question is if there is specific benefit from common abstraction. So we could have something like

    DeferredValue<T>

with method like

    T resolve(CommandContext<?> ctx)

(or such)
but I am not sure what immediate value there'd be for handling these things (SortClause would be next immediate choice) in common way using super type.

But I'll think about this a bit, in case something comes up.

Reply:
Refactored to remove all processing logic from here, move to `CommandContext`. Also based on `JsonDefinition`, so this wrapper only needed for annotations, basically.